source/developers/projects/engine/api/script-sandbox-configuration.rst

```diff
@@ -81,3 +81,101 @@ It is possible to completely disable the Groovy sandbox for all scripts. To disa
+
+One limitation is that an exception is thrown during execution when a Groovy class has a property and a getter method for the property.  Here's an example code that throws an exception during execution:
+
+   .. code-block::
+
+      class Test {
+        private String message
+
+        public String getMessage() {
+           return this.message
+        }
+      }
+
+      def t = new Test()
+      t.message = "this is a test"
+
+      return t.getMessage()
+
+   |
+
+Here's the error thrown in the logs by the code above:
+
+.. code-block:: text
+
+   Caused by: java.lang.StackOverflowError
+	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:693)
+	at groovy.lang.GroovyClassLoader$InnerLoader.loadClass(GroovyClassLoader.java:450)
+	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:812)
+	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:800)
+	at sun.reflect.GeneratedMethodAccessor340.invoke(Unknown Source)
+	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
+	at java.lang.reflect.Method.invoke(Method.java:498)
+	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
+	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
+	at groovy.lang.MetaClassImpl.getProperty(MetaClassImpl.java:1845)
+	at groovy.lang.MetaClassImpl.getProperty(MetaClassImpl.java:3773)
+	at Test.getProperty(test.get.groovy)
+	at org.codehaus.groovy.runtime.InvokerHelper.getProperty(InvokerHelper.java:190)
+	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.getProperty(ScriptBytecodeAdapter.java:469)
+	at org.kohsuke.groovy.sandbox.impl.Checker$7.call(Checker.java:392)
+	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onGetProperty(GroovyInterceptor.java:68)
+	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onGetProperty(SandboxInterceptor.java:297)
+	at org.kohsuke.groovy.sandbox.impl.Checker$7.call(Checker.java:390)
+	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:394)
+	at org.kohsuke.groovy.sandbox.impl.Checker$checkedGetProperty$1.callStatic(Unknown Source)
+	at Test.getMessage(test.get.groovy:5)
+
+|
+
+**Workarounds**
+
+There are a couple of things you can do to get around the exception being thrown:
+
+* Do not use getter methods and instead access the property directly |br|
+  Using the example above, we'll access the property directly:
+
+     .. code-block::
+
+        class Test {
+          private String message
+        }
+
+        def t = new Test()
+        t.message = "this is a test"
+
+        return t.message
+
+     |
+
+* Use a different name for the property and the getter method |br|
+  Again, using the example above, we'll use a different name from the property for the getter method:
+
+     .. code-block::
+
+        class Test {
+          private String message
+
+          public String getTheMessage() {
```

Comment:
Please change the name of the property instead of the getter, just for consistency with coding standards